### PR TITLE
Implement decimal and decimal128

### PR DESCRIPTION
**Pull request by bterlson**
Two additional things that weren't covered in the design meeting (see #1733):

1. In order to handle numeric literals of any precision, I've added `valueAsString` to both numeric literal nodes and numeric literal types. This way, compilers and emitters can use either depending on the situation.
2. The POR was decimal extends numeric, but decimal types are a kind of floating point. I have kept with the POR because I believe `float` in common parlance refers to a binary floating point number exclusively.
3. OpenAPI has added decimal and decimal128 to the [format registry](https://spec.openapis.org/registry/format/), seems like no harm in using them.

Todo is write some more tests.

### CODE REVIEW

File: docs/language-basics/type-literals.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ two
 
 ## Numeric literal
 
-Numeric literals can be declare by using the raw number (`interger` or `float`)
+Numeric literals can be declare by using the raw number
```
Comment (markcowl) [inline docs/language-basics/type-literals.md:67]:
```suggestion
Numeric literals can be declared by using the raw number
```

File: packages/compiler/test/checker/relation.test.ts [hunk 2]
```diff
@@ -446,6 +450,29 @@ describe("compiler: checker: type relations", () => {
...
 
+  describe("decimal target", () => {
+    it("can assign decimal", async () => {
+      await expectTypeAssignable({ source: "decimal", target: "decimal" });
+    });
+    it("can assign decimal128", async () => {
+      await expectTypeAssignable({ source: "decimal128", target: "decimal" });
+    });
+    it("can assign numeric literals", async () => {
+      await expectTypeAssignable({ source: "-2147483448", target: "decimal" });
```
Comment (markcowl) [inline packages/compiler/test/checker/relation.test.ts:461]: Should we add digits beyond the decimal point? (e.g. -5.50, 10.25)

File: docs/language-basics/built-in-types.md [hunk 3]
```diff
@@ -11,22 +11,24 @@ Built in types are related to each other according to the rules described in [ty
...
+| `uint32`     | `0` to `4,294,967,295`                                                                                       | Unsigned 32-bit integer                   |
+| `uint16`     | `0` to `65,535`                                                                                              | Unsigned 16-bit integer                   |
+| `uint8`      | `0` to `255 `                                                                                                | Unsigned 8-bit integer                    |
+| `float32`    | <code> ±1.5 x 10<sup>45</sup></code> to <code>±3.4 x 10<sup>38</sup></code>                                  | A 32 bit floating point number            |
+| `float64`    | <code>±5.0 × 10<sup>−324</sup></code> to <code>±1.7 × 10<sup>308</sup></code>                                | A 64 bit floating point number            |
+| `decimal`    |                                                                                                              | A decimal number                          |
```
Comment (markcowl) [inline docs/language-basics/built-in-types.md:30]: Feels like we need something here to indicate that this is a fixed point number and should not be used to represne floating point numbers.

**Review by markcowl — DISMISSED** [review]
A couple of minor nits, otherwise great

File: packages/compiler/test/checker/relation.test.ts (hunk 2 above)
Comment (bterlson) [reply packages/compiler/test/checker/relation.test.ts:461]: Good call, I'll add one

File: docs/language-basics/built-in-types.md (hunk 3 above)
Comment (bterlson) [reply docs/language-basics/built-in-types.md:30]: Decimals are actually floating point, but I felt that including that fact in the description would just confuse matters (this is related to point #2 above).

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**